pymeasure/adapters/adapter.py

```diff
@@ -114,6 +119,33 @@ def binary_values(self, command, header_bytes=0, dtype=np.float32):
         raise NameError("Adapter (sub)class has not implemented the "
                         "binary_values method")
 
+    def __getattribute__(self, name):
+        """ Override to acquire the 'connection_lock' lock through the duration of attribute access.
+        """
+        if not name == "connection_lock" and \
+                object.__getattribute__(self, "connection_lock_initialized"):
+            with object.__getattribute__(self, "connection_lock"):
+                ret = object.__getattribute__(self, name)
+        else:
+            ret = object.__getattribute__(self, name)
+
+        return ret
+
+    def __setattr__(self, name, value):
+        """ Override to acquire the 'connection_lock' lock through the duration of attribute setting.
+        """
+        if not name == "connection_lock" and \
+                object.__getattribute__(self, "connection_lock_initialized"):
+            with object.__getattribute__(self, "connection_lock"):
+                object.__setattr__(self, name, value)
```

Comment:
```suggestion
                setattr(self, name, value)
```

Reply:
In this case, using `setattr` leads to `RecursionError: maximum recursion depth exceeded` since everytime `setattr` is called, the overridden version of `__setattr__` is called. Similar to in the new version of `__getattribute__` I believe we should use `super().__setattr__()`.

Reply:
Indeed. Sorry I overlooked this.